src/mongoose_tcp_listener.erl

```diff
@@ -115,34 +115,54 @@ accept_loop(ListenSocket, Module, Opts = #{proxy_protocol := ProxyProtocol}) ->
             ?MODULE:accept_loop(ListenSocket, Module, Opts)
     end.
 
--spec do_accept(gen_tcp:socket(), boolean()) ->
+-spec do_accept(gen_tcp:socket(), options()) ->
     {ok, gen_tcp:socket(), connection_details()} | {error, term()}.
-do_accept(ListenSocket, ProxyProtocol) ->
+do_accept(ListenSocket, Opts) ->
     case gen_tcp:accept(ListenSocket) of
-        {ok, Socket} when ProxyProtocol ->
-            read_proxy_header(Socket);
         {ok, Socket} ->
-            {ok, {DestAddr, DestPort}} = inet:sockname(Socket),
-            {ok, {SrcAddr, SrcPort}} = inet:peername(Socket),
+            ?MODULE:read_connection_details(Socket, Opts);
```

Comment:
Why do we use `?MODULE:` to call the function from the same module? Is this function not always available?

Reply:
Thx